### PR TITLE
Add consensus genome workflow

### PR DESCRIPTION
**Pull request by tfrcarvalho**
* Adds consensus genome workflow with corresponding Dockerfile

### Notes

A few minor things to fix(?) in the future:
* extra required variables that this workflow would not need
* filename of the workflow for single-wdl SFN has to be `run`
	* might be confusing in the future with many such workflows
    * downloaded WDL filename is not very informative  
* repo-wide version

### CODE REVIEW

File: consensus-genome/run.wdl [hunk 1]
```diff
@@ -0,0 +1,690 @@
...
+
+        String docker_image_id
+    }
+
+    command <<<
+        samtools index ${bam}
+        samtools mpileup -A -d "~{mpileupDepth}" -Q0 "~{bam}" | ivar consensus -q "~{ivarQualThreshold}" -t "~{ivarFreqThreshold}" -m "~{minDepth}" -n N -p "~{prefix}primertrimmed.consensus"
+        echo ">""~{sample}" > "~{prefix}consensus.fa"
+        seqtk seq -l 50 "~{prefix}primertrimmed.consensus.fa" | tail -n +2 >> "~{prefix}consensus.fa"
+
```
Comment (jshoe) [inline consensus-genome/run.wdl:432]: nit: extra line break

File: consensus-genome/run.wdl [hunk 2]
```diff
@@ -0,0 +1,690 @@
...
+        File fastqs_0
+        File fastqs_1
+        File ref_fasta
+
+        String no_reads_quast
+
+        Int threads = 4
+
+        String docker_image_id
+
```
Comment (jshoe) [inline consensus-genome/run.wdl:457]: nit: extra blank line?

**Comment by katrinakalantar** [comment]
I would also like to ensure that we credit initial repo from which this code was based here: https://github.com/czbiohub/sc2-illumina-pipeline

File: consensus-genome/run.wdl [hunk 3]
```diff
@@ -0,0 +1,690 @@
...
+        # all of the following seem to be parameters in params.* of the nextflow pipeline;
+        # could they be extracted to a separate params file? put in .json input to wdl?
+        Boolean trim_adapters = true
+
+        Float   intrahost_min_frac = 0.5
+        String  intrahost_ploidy   = "2"
+        Boolean intrahost_variants = true
+
+        Float ivarFreqThreshold = 0.6
```
[the remark this answers is not in the document]
Comment (katrinakalantar) [reply consensus-genome/run.wdl:32]: specifically, update to `ivarFreqThreshold = 0.9`

**Comment by tfrcarvalho** [comment]
> I would also like to ensure that we credit initial repo from which this code was based here: [czbiohub/sc2-illumina-pipeline](https://github.com/czbiohub/sc2-illumina-pipeline)

Added a comment at the top of the workflow file.

File: consensus-genome/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,37 @@
+FROM ubuntu:18.04
+ARG DEBIAN_FRONTEND=noninteractive
+
+LABEL maintainer="IDseq Team idseq-tech@chanzuckerberg.com"
+LABEL description = "Image for consensus genome by metagenomic sequencing with spiked primer enrichment"
```
[the remark this answers is not in the document]
Comment (tfrcarvalho) [reply consensus-genome/Dockerfile]: Unfortunately there seems to be an incompatibility on the conda packages and I am unable to install that version at this time. 
Will work on alternatives on a future PR.

**Pull request merged**